### ISSUE TITLE
In memory security contexts are not supported

### ISSUE THREAD

**Issue by riedel** [dask/dask-jobqueue#520]
See https://github.com/dask/distributed/blob/3754a6694f72207b69aaf661445aecf18fb5f0cb/distributed/security.py#L211

i.e. one cannot use Security.temporary() to create a cluster 

distrinuted allows using in memory keys where the content is in memory. It does so by checking if the value is multiline.

the expansion to the extra arguments does not support this:
https://github.com/dask/dask-jobqueue/blob/79a9a3bee55c56195f97b3e70bda15c12c3b19be/dask_jobqueue/core.py#L224

in this case a temporary file needs to be created

**Comment by riedel** [comment]
Just tracking this because I will need to fix this in #519 to work on #496

**Comment by riedel** [comment]
I guess the main challenge will be handling the temporary files as they need to survive until the cluster is closed. Maybe also writing the whole cert into the arg works...

**Comment by riedel** [comment]
>  And like @jacobtomlinson said we are assuming a shared file system anyways.

I was actually thinking this also the whole time until you reminded me of `/tmp`. I started digging into the code to find where this assumption is actually encoded. It seems that HTCondor is explicit about this:
https://github.com/dask/dask-jobqueue/blob/26fd3ae8e8c18d26f5e203f2c7f1a1be91e113d4/dask_jobqueue/htcondor.py#L236

However, when setting up the new CI I think I completely ignored this:
https://github.com/dask/dask-jobqueue/blob/9944654374a4e06d3a3289069d2ec0a61ae96162/ci/htcondor/docker-compose.yml#L56-L57

So I think it might probably not depend on this.

Here is a hint that this can be very different on different setups:
https://github.com/dask/dask-jobqueue/blob/1f9ae1ecc79a5b76930e56c8801f3b8ace659877/dask_jobqueue/lsf.py#L201

Here you find hints that even python might not be shared:
https://github.com/dask/dask-jobqueue/blob/58e0ce2111e74484dca4e2469698bbe68a2240b2/docs/source/debug.rst#checking-job-script

Bottom line:  the testing of security should actually be done on all schedulers and the assumption should be dropped, IMHO. It might well be the case that also file based certificates need to be copied to the worker.#524 

What do you think.

@guillaumeeb : thanks for nagging me here. I think this led to much clearer thinking. One positive outcome should be that the testing strategy should be overhauled a bit, since the CI run on #519 for all the schedulers should be a warning...

**Comment by riedel** [comment]
I confirmed that actually file based certificates also won't work on the HTCondor and slurm CI: #532 . Now the price question: is this a bug of the ci or the code. It should be at least document that one has to be smart enough to put the certificates in some accessible place. On the other hand: how would code fixing this issue know where this is...

**Comment by guillaumeeb** [comment]
Sorry for the delay @riedel, and as always thanks for your investments in dask-jobqueue!

> Now the price question: is this a bug of the ci or the code

I'd said neither.

I think all dask-jobqueue Cluster can work without any shared file system in some specific setups. Some CI environments show this. It should remain the same.

The code for generating a temporary file should just try to do it in the user folder, and be explicit (with some logging?) that the place where the file is generated should be accessible from all workers. Maybe we can add a configuration option for this location (as we do for worker log directory). All this should also be clearly documented.

For the CI, we could test this only in CI environment with shared folders. I don't know if we want to add a shared file system for all tests, just to test this. I don't think so.
